src_c/rect.c

```diff
@@ -357,6 +357,48 @@ _pg_do_rects_intersect(SDL_Rect *A, SDL_Rect *B)
             MAX(A->y, A->y + A->h) > MIN(B->y, B->y + B->h));
 }
 
+static int
+_pg_do_normalized_rects_intersect(SDL_Rect *A, SDL_Rect *B)
+{
+    // A.left   < B.right  &&
+    // A.top    < B.bottom &&
+    // B.left   < A.right  &&
+    // B.top    < A.bottom
+
+    return (A->x < B->x + B->w && A->y < B->y + B->h && B->x < A->x + A->w &&
```

Comment:
Wouldn't something like this be faster?
` return not (a.left >= b.right or a.top >= b.bottom or b.left >= a.right or b.top >= a.bottom)` 
I mean it would short circuit probably half of the time, but there is an additional 'no'. My guess would be it takes half as long in the mean, but maybe I'm wrong (due to overhead or something). Only a measurement will show the difference.

Reply:
here my test code:
```
from pygame import Rect

rect = Rect(0, 0, 10, 10)
rects = [Rect(x, y, 5, 5) for x in range(-10, 15) for y in range(-10, 15)]


def collide1(a, b):
    return a.left < b.right and a.top < b.bottom and b.left < a.right and b.top < a.bottom


def collide2(a, b):
    return not (a.left >= b.right or a.top >= b.bottom or b.left >= a.right or b.top >= a.bottom)


for r in rects:
    c1 = collide1(rect, r)
    c2 = collide2(rect, r)
    c3 = rect.colliderect(r)
    if c1 != c2:
        print(f"c1 {c1} and c2 {c2} differ: {rect} vs {r} and c3: {c3}")

```

Reply:
> Wouldn't something like this be faster? ` return not (a.left >= b.right or a.top >= b.bottom or b.left >= a.right or b.top >= a.bottom)` I mean it would short circuit probably half of the time, but there is an additional 'not'.

it now also has to calculate the "=" part in <= for each coord. Who knows i'll test that. thanks for the review tho!

Reply:
There's not much of a difference. Not that i could measure. One test mught be faster, the other slower (all with the same rect and pos), so it's in the noise.

Reply:
ok, thanks for testing! In this case I would leave it as it is.